### DISCUSSION TITLE
Migrating events from non-MISP tool into MISP

### DISCUSSION THREAD

**Discussion by GDAdamSiwek** [MISP/PyMISP#785] [General]
I am new to MISP, am curious if I can migrate events from a legacy tool into MISP, and can the attributes from the legacy events become custom attributes on a MISP event?

Any help appreciated!

**Comment by Rafiot** [comment]
Can you give us a sample? I'm not sure to understand what you mean.

**Comment by GDAdamSiwek** [comment]
Here is a subset of the columns of an event from my legacy:

1. id
2. eval_date
3. date_created
4. date_updated
5. indicator_type

Can I create an event via PyMISP, and pin these attributes to it?

**Reply by Rafiot** [reply]
I'd go for an object with each of these attributes, this is probably the easiest.

**Comment by GDAdamSiwek** [comment]
So,  the procedure for this is:

1. Create Object template
2. Register Template with MISP instance
3. Create event, object and load legacy attributes into object
4. Attach object to event

?

**Reply by Rafiot** [reply]
Pretty much yes. You can also just create the template in the subdirectory where all the other templates are in PyMISP and use it directly from there.

The question you need to answer is how you want to add them into event(s): one event with all the data in objects? One event/month? One event/day? It will depend how many objects you end-up having, and what you want to do with the data.

**Comment by GDAdamSiwek** [comment]
Much THANKS for that info, but I need further clarity.  The five fields I mentioned before, all five go into a single object or there are five objects, one per field?

________________________________

**Reply by Rafiot** [reply]
an object contains attributes, so you will have N attributes in your object. One other thing you might want to do is to use the first_seen/last_seen field in the object itself to represent created and updated, if it makes sense.

The other thing I'm not clear about is where the actual indicator is? You're talking of indicator type, but not the indicator itself.

**Comment by GDAdamSiwek** [comment]
FYI,
    I've inherited this task( migration to MISP), so I might not have the vocabulary right about indicator types and more. But thanks for the feedback!

________________________________

**Comment by Rafiot** [comment]
You're welcome! If you need further help, don't hesitate. And if you can share an actual sample, it might make things easier.

**Comment by GDAdamSiwek** [comment]
I'm not able to share sample data anytime soon, but , shifting lanes, I have a question about searches of MISP object.  Can the web UI perform queries on the custom object attributes?

________________________________

**Reply by Rafiot** [reply]
The attributes aren't custom, only the objects containing standard attributes are, so answer is yes.

**Comment by GDAdamSiwek** [comment]
Raphaël,

    The template file to create a custom object exists and resides in the objects/templates directory.  But how do enforce the template during MISP object creation?

________________________________

**Comment by GDAdamSiwek** [comment]
Raphaël,
  OK,  I created the Object template file, and located it in the MISP misp_object template directory.  But how do I enforce that template during object creation?

**Comment by Rafiot** [comment]
There are a few moving parts, so it's a bit hard to answer. As you have a custom template, I'm assuming that you installed PyMISP from the repository.

You can look at this tutorial: https://github.com/MISP/PyMISP/blob/main/docs/tutorial/FullOverview.ipynb (search for `
Use locally defined object templates`).
